core/validation/failures.py

```diff
@@ -267,6 +267,11 @@ def to_message(self) -> tuple[str, str, str, str]:
         # additional logic for outcomes to differentiate between footfall and non-footfall
         if sheet == "Outcomes" and self.row_values[4] == "Year-on-year % change in monthly footfall":
             section = "Footfall Indicator"
+            # +5 as GeographyIndicator is 5 rows below Footfall Indicator
+            if column == "Geography Indicator":
+                actual_index = self.row_indexes[0] + 5
+                cell = "C" + str(actual_index)
```

Comment:
nit picky but I prefer an fstring here? I think it means we can do away with the `str` function

Reply:
Then as per my other comment this would be modified to:

```
@dataclass
class InvalidEnumValueFailure(ValidationFailure):
    """Class representing an invalid enum value failure."""

    sheet: str
    column: str
    row_indexes: list[int]
    row_values: tuple
    value: Any

    def to_message(self) -> tuple[str, str, str, str]:
        sheet = INTERNAL_TABLE_TO_FORM_TAB[self.sheet]
        column, section = INTERNAL_COLUMN_TO_FORM_COLUMN_AND_SECTION[self.column]
        message = msgs.DROPDOWN
        
        cell_index = construct_cell_index(table=self.sheet, column=self.column, row_indexes=self.row_indexes)

        # additional logic for outcomes to differentiate between footfall and non-footfall
        if sheet == "Outcomes" and self.row_values[4] == "Year-on-year % change in monthly footfall":
            cell_index = construct_cell_index(
                table=self.sheet,
                column=self.column,
                row_indexes=self.row_indexes,
                additional_lookups=("Footfall Indicator",),
            )

        # additional logic for risk location
        if sheet == "Risk Register":
            project_id = self.row_values[1]
            section = risk_register_section(project_id, self.row_indexes[0], self.sheet)

        if section == "Project Funding Profiles":
            project_number = get_project_number_by_position(self.row_indexes[0], self.sheet)
            section = f"Project Funding Profiles - Project {project_number}"

        return sheet, section, cell_index, message
```

Reply:
Not convinced this is the right solution.. or even worth making this additional complexity for. It all stems from having two different tables in the spreadsheet feed into the same dataframe.. one way to fix this would be to have separate dataframes for footfall and non-footfall

Reply:
This would actually be quite simple to do because we'd just create a new mapping for the footfall table that maps to the same db model:

```
DataMapping(
    worksheet_name="Outcome_Data",  # THIS WOULD BE e.g. Footfall_Outcome_Data
    model=ents.OutcomeData,
    column_mapping={
        "Submission ID": "submission_id",
        "Project ID": "project_id",
        "Programme ID": "programme_id",
        "Outcome": "outcome",
        "Start_Date": "start_date",
        "End_Date": "end_date",
        "UnitofMeasurement": "unit_of_measurement",
        "GeographyIndicator": "geography_indicator",
        "Amount": "amount",
        "Actual/Forecast": "state",
        "Higher Frequency": "higher_frequency",
    },
    fk_relations=[
        ("submission_id", ents.Submission, "submission_id", "submission_id"),
        ("programme_id", ents.Programme, "programme_id", "programme_id"),
        ("project_id", ents.Project, "project_id", "project_id"),
        ("outcome_name", ents.OutcomeDim, "outcome_id", "outcome"),
    ],
),
```

Reply:
Really thinking out loud now... but another example of where this could be useful is for `Other Funding Sources`.. it preserves more context of the original format which would then allow us to use the standard validation in more cases as well. i.e. The "Funding Source" and "Secured" dropdowns require validation for Other Funding Sources but not for the standard TF Funding Sources